File: src/main/java/com/github/simiacryptus/aicoder/openai/CompletionResponse.java
package com.github.simiacryptus.aicoder.openai;

import org.jetbrains.annotations.NotNull;

import java.util.Arrays;
import java.util.Optional;

public class CompletionResponse {
    @SuppressWarnings("unused")
    public String id;
    @SuppressWarnings("unused")
    public String object;
    @SuppressWarnings("unused")
    public int created;
    @SuppressWarnings("unused")
    public String model;
    public Choice[] choices;
    @SuppressWarnings("unused")
    public ApiError error;

    @SuppressWarnings("unused")
    public Usage usage;

    public CompletionResponse() {
    }

    public CompletionResponse(String id, String object, int created, String model, Choice[] choices, ApiError error) {
        this.id = id;
        this.object = object;
        this.created = created;
        this.model = model;
        this.choices = choices;
        this.error = error;
    }

    public @NotNull Optional<CharSequence> getFirstChoice() {
        return Optional.ofNullable(this.choices).flatMap(choices -> Arrays.stream(choices).findFirst()).map(choice -> choice.text.trim());
    }
}

File: src/main/kotlin/com/github/simiacryptus/openai/CompletionResponse.kt
package com.github.simiacryptus.openai

import java.util.*

class CompletionResponse {
    @Suppress("unused")
    var id: String? = null

    @Suppress("unused")
    var `object`: String? = null

    @Suppress("unused")
    var created = 0

    @Suppress("unused")
    var model: String? = null
    var choices: Array<CompletionChoice> = arrayOf()

    @Suppress("unused")
    var error: ApiError? = null

    @Suppress("unused")
    var usage: Usage? = null

    constructor()
    constructor(
        id: String?,
        `object`: String?,
        created: Int,
        model: String?,
        choices: Array<CompletionChoice>,
        error: ApiError?
    ) {
        this.id = id
        this.`object` = `object`
        this.created = created
        this.model = model
        this.choices = choices
        this.error = error
    }

    val firstChoice: Optional<CharSequence>
        get() = Optional.ofNullable(choices).flatMap { choices: Array<CompletionChoice>? ->
            Arrays.stream(
                choices
            ).findFirst()
        }.map { choice: CompletionChoice -> choice.text!!.trim { it <= ' ' } }
}